src/Engine/GameEvents/GameEventListener.java

```diff
@@ -90,4 +91,31 @@ default public void unregister(GameInstance gi)
   default public GameEventQueue receiveMassDamageEvent(Commander attacker, Map<Unit, Integer> lostHP){ return null; };
   default public GameEventQueue receiveModifyFundsEvent(Army beneficiary, int fundsDelta){ return null; };
 
+  /**
+   * Want to be lazy, but too lazy to implement laziness? This class is for you.
+   */
+  public static interface CacheInvalidationListener extends GameEventListener
```

Comment:
This... really isn't less work than just doing it in MTDA directly the way it was before. So far as I can tell, it is strictly more work since you still need all the same calls, and it's less intuitive to boot.

Reply:
Okay, but:
It's less likely to get out of date (it was already missing some event hooks).
I can re-use that work elsewhere, making it less work overall.